ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/CardinalClient.kt

```diff
@@ -0,0 +1,136 @@
+package com.braintreepayments.api.threedsecure
+
+import android.content.Context
+import com.braintreepayments.api.core.BraintreeException
+import com.braintreepayments.api.core.Configuration
+import com.cardinalcommerce.cardinalmobilesdk.Cardinal
+import com.cardinalcommerce.cardinalmobilesdk.enums.CardinalEnvironment
+import com.cardinalcommerce.cardinalmobilesdk.enums.CardinalRenderType
+import com.cardinalcommerce.cardinalmobilesdk.enums.CardinalUiType
+import com.cardinalcommerce.cardinalmobilesdk.models.CardinalChallengeObserver
+import com.cardinalcommerce.cardinalmobilesdk.models.CardinalConfigurationParameters
+import com.cardinalcommerce.cardinalmobilesdk.models.ValidateResponse
+import com.cardinalcommerce.cardinalmobilesdk.services.CardinalInitService
+import org.json.JSONArray
+
+internal class CardinalClient {
+    var consumerSessionId: String? = null
+        private set
+
+    @Throws(BraintreeException::class)
+    @Suppress("TooGenericExceptionCaught")
+    fun initialize(
+        context: Context,
+        configuration: Configuration,
+        request: ThreeDSecureRequest,
+        callback: CardinalInitializeCallback
+    ) {
+        configureCardinal(context, configuration, request)
+
+        try {
+            val cardinalInitService = object : CardinalInitService {
+                override fun onSetupCompleted(sessionId: String) {
+                    consumerSessionId = sessionId
+                    callback.onResult(consumerSessionId, null)
+                }
+
+                override fun onValidated(validateResponse: ValidateResponse?, serverJWT: String?) {
+                    if (consumerSessionId == null) {
+                        callback.onResult(
+                            consumerSessionId = null,
+                            error = BraintreeException("consumer session id not available")
+                        )
+                    } else {
+                        callback.onResult(consumerSessionId, null)
+                    }
+                }
+            }
+
+            Cardinal.getInstance()
+                .init(configuration.cardinalAuthenticationJwt, cardinalInitService)
+        } catch (e: RuntimeException) {
+            throw BraintreeException("Cardinal SDK init Error.", e)
+        }
+    }
+
+    @Throws(BraintreeException::class)
+    @Suppress("TooGenericExceptionCaught")
+    fun continueLookup(
+        threeDSecureParams: ThreeDSecureParams,
+        challengeObserver: CardinalChallengeObserver?
```

Comment:
Is it possible to make the `challengeObserver` param non-optional?

Reply:
I originally tried to make this parameter non-null, but then the null check needed to be moved to `ThreeDSecureActivity` since the observer is set in the Activity's onCreate. I felt that it made more sense to keep the logic out of the Activity which made it much easier to add a unit test covering this path. But happy to revisit if folks feel differently!

Reply:
Ok got it 👍 that makes sense.